### DISCUSSION TITLE
A widget showing a list of revisions (for Authors)

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Revisions#908] [Feature Requests]
> A widget displaying “Pending (4)” which simply shows the number of pending revisions to your posts with a link to those revisions. This would make it easier for users (authors) to keep track of revisions they should tend to so that the admins aren’t doing all the work.

**Comment by JoshJourney** [comment]
Pinch me. I'm dreaming! :+1:  A small widget that would make a huge impact on the workflow. For years debated "private content or wiki". Revisionary will enable the best of both worlds by giving post author's the ability to take in contributions from the community. Once this widget exists I'd be more than happy to donate code to make it look visually dazzling as well as an easy restriction method for communities. For example restrict members from site extras if they let the revision counter get too high to keep them accountable by storing the value in a variable and creating restriction filters. Again only negligent users. ;-)

**Comment by stevejburge** [comment]
Thanks @Drumbell. Thinking this through ... the Widget will have a list of revisions. Maybe it's configurable to show past or upcoming revisions?

I think the original user request was for upcoming revisions.

But I can see a use-case for past revisions too, similar a Wikipedia page history

![Screen Shot 2019-08-27 at 11 38 18 AM](https://user-images.githubusercontent.com/3868028/63786020-4f85fa80-c8bf-11e9-87f2-0361a80d1d7d.png)

**Comment by JoshJourney** [comment]
Interesting idea. Here's a mockup showing a "global revisions" for that user:

![Revision Widget](https://user-images.githubusercontent.com/4361528/63792978-8d7d2180-c8b3-11e9-91fa-025571a0b8c6.png)

Perhaps a second icon could appear at the top that shows only content related to the current post viewed. Imagine what is in your screen shot being placed in a second icon dropdown menu for that post. This would allow easy history comparing for specific posts; albeit perhaps with a leaner display showing the date, compare, undo, and change count if it's not too tricky.

**Comment by stevejburge** [comment]
Thanks @Drumbell. This is super-interesting. 

Can I ask a key question ... Who is the intended user of this widget? Is it content creators on this WordPress site? I ask because I can imagine Wikipedia-style uses too where you can show visitors a history of the content.

**Comment by JoshJourney** [comment]
> Can I ask a key question ... Who is the intended user of this widget? Is it content creators on this WordPress site?

Good question. The following widget is proposed for post authors to attend received pending revisions to their posts. This would prevent users from "missing a beat" when getting lots of emails.

![Compare Button](https://user-images.githubusercontent.com/4361528/63815235-b4a21600-c8e8-11e9-80fc-8113683a58d1.png)

Josh is a post author. Daniel is a member who signed up for the site, saw a potential addition, made a revision. Josh clicks the "new-revisions" counter, clicks green  compare button. See's highlighted changes, likes them, hits approve. This would make the workflow ridiculously easy/fast for everyone including admins and encourage lots of engagement.

If Revisionary goes [beyond revisor role](https://github.com/publishpress/Revisionary/issues/6) then this would be for all registered users.

> I ask because I can imagine Wikipedia-style uses too where you can show visitors a history of the content.

This would be another fantastic addition that would rally up content crafters and make easier accountability. Aside from well managed article staffs, wiki lovers would greatly benefit from this. I'm proposing a second icon called "post-history" beside the icon "my-revisions":

![Revisions dropdown](https://user-images.githubusercontent.com/4361528/63814760-d0a4b800-c8e6-11e9-88a5-5f8fcce7b460.png)

This version includes an explanation header for the dropdown:
![Screenshot from 2019-08-27 16-08-02](https://user-images.githubusercontent.com/4361528/63815624-29c21b00-c8ea-11e9-86fb-deae82178e83.png)


The counter specifically targets pending revisions to the current viewed post. Huge thanks for continuing the work on Revisionary.

**Comment by JoshJourney** [comment]
Created a 1 minute video showing how it could work:
https://www.youtube.com/watch?v=t_w0Jionek8

At the end I was showing when comparing versions there isn't an easy way to publish from versions view. Rather than replacing the button text shown in video I should have copied it and rename it to "Publish Now" which could enable users who like the changes to quickly publish from the differences page. Notice how it took seconds for the post author to go from "not aware of revisions" to being able to see revisions, know changed, and approve them.

**Comment by stevejburge** [comment]
Thanks so much for this @Drumbell 

This idea won't make the first Revisionary 2.0 (that's close to being locked in) now, but it would be a great follow-up feature.

I've spun off the widget for visitors into a separate issue: https://github.com/publishpress/Revisionary/issues/15

**Comment by JoshJourney** [comment]
I appreciate it. In the mean time perhaps with a snippet of php one could count the number of revisions the current author should attend? I'd put the following in a custom-template header.php:
`if($revisionCount > 0) {
    echo "<a href=\"wp-admin/admin.php?page=revisionary&view=revisionqueue&filter=mine\">";
    echo "<span>" . $revisionCount . "</span></a>";
}`

I'd add classes, stylize it, ect.

**Comment by stevejburge** [comment]
Thanks @Drumbell. If you want a look at Revisionary Pro, feel free email our support. That has a new "Mine" tab in the admin

**Comment by JoshJourney** [comment]
Request send.

> That has a new "Mine" tab in the admin

That's why my "somewhat accurate" snippet of PHP referenced "wp-admin/admin.php?page=revisionary&view=revisionqueue&filter=mine". Made a few guesses with the variable parameters for accessing the "my queue" tab. This will be a much appreciated filter. While I'm aware that the backend will be needed, users will spend the majority of their time on the frontend which is why I'm trying to have a way for users to easily see when there is pending revisions to their posts on the frontend. Adding a frontend snippet showing the counter will make those who care about their content much more responsive.
